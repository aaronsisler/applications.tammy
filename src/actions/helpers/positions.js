export const setPositions = (positions) => ({
    type: 'SET_POSITIONS',
    positions
});
