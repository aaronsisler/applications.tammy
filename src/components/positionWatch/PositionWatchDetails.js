import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import history from 'Tools/history';
import LinkWrapper from 'Universal/LinkWrapper';
import PositionDetailsContent from 'Position/PositionDetailsContent';
import PositionWatchEditWidget from 'PositionWatch/PositionWatchEditWidget';

export class PositionWatchDetails extends React.Component {
    constructor(props) {
        super(props);
    }

    handleNavigateBack = () => history.push('/dashboard');

    renderNoWatchedPosition = () => (
        <div className="inbox_details__empty inbox_mobile">
            Position is not currently being watched. Please make sure to add a watch.
        </div>
    )

    renderNoPosition = () => (
        <div className="inbox_details__empty inbox_mobile">
            Please select an item to view
        </div>
    )

    render() {
        const { position, positionId } = this.props;
        if (this.props.positionId && !this.props.position) {
            return this.renderNoWatchedPosition();
        }

        if (!this.props.positionId) {
            return this.renderNoPosition();
        }

        return (
            <div className="inbox_details" >
                <div className="inbox_details_header">
                    <div className="inbox_details_header__actions">
                        <button
                            className="inbox_details_header__mobile_button"
                            onClick={this.handleNavigateBack}
                        >
                            Back to List
                        </button>
                        <LinkWrapper
                            linkText='View Applicants'
                            to={`/applicants/${positionId}`}
                        />
                    </div>
                    <div className="inbox_details_header__content">
                        <div className="inbox_details_header__title">
                            {position.title}
                        </div>
                        <div className="inbox_details_header__job_id">
                            Job Id: {position.jobId}
                        </div>
                        <div className="inbox_details_header__location">
                            Location: {position.location}
                        </div>
                    </div>
                </div>
                <div className="inbox_details_content">
                    <PositionWatchEditWidget positionId={this.props.positionId} />
                    <PositionDetailsContent position={position} />
                </div>
            </div>
        );
    }
}

/* istanbul ignore next */
const mapStateToProps = (state, props) => {
    const { positionId } = props.match.params;
    const position = positionId
        ? state.positionsWatched.find((statePosition) => statePosition.positionId == positionId)
        : undefined;

    return ({
        position,
        positionId,
    });
};

export default connect(mapStateToProps)(PositionWatchDetails);

PositionWatchDetails.propTypes = {
    position: PropTypes.object,
    positionId: PropTypes.string,
};
