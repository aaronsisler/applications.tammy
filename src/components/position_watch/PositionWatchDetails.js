import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import PositionWatchSubscriptionLevel from './PositionWatchSubscriptionLevel';
import PositionDetailsContent from '../shared/position/PositionDetailsContent';

export class PositionWatchDetails extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        const { position, positionWatched } = this.props;
        return (
            <div className="position_watch_details">
                {!position &&
                    <div className="empty">
                        Please select an item to view
                    </div>
                }
                {position &&
                    <div className="position_watch_details_widget">
                        <div className="position_watch_details_header">
                            <div>
                                <div className="position_watch_details__title">
                                    {position.title}
                                </div>
                                <div className="position_watch_details__job_id">
                                    Job Id: {position.jobId}
                                </div>
                            </div>
                            <div className="position_watch_details__subscription_level">
                                <PositionWatchSubscriptionLevel
                                    id={positionWatched.id}
                                    subscriptionLevel={positionWatched.subscriptionLevel}
                                />
                            </div>
                        </div>
                        <div className="position_watch_details_link">
                            <Link
                                className="nav_link"
                                to={`position/${position.id}`}
                            >
                                Go to Position Workflow
                            </Link>
                        </div>
                        <PositionDetailsContent position={position} />
                    </div>
                }
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    position: state.position,
    positionWatched: state.positionWatch.position,
});

export default connect(mapStateToProps)(PositionWatchDetails);


PositionWatchDetails.propTypes = {
    position: PropTypes.object,
    positionWatched: PropTypes.object,
};

