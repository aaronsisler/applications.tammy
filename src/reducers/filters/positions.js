const positionsFilterReducerDefaultState =
{
    text: '',
};

const positionsFilterReducer = (state = positionsFilterReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_POSITIONS_TEXT_FILTER':
            return {
                ...state,
                text: action.text
            };
        case 'CLEAR_POSITIONS_TEXT_FILTER':
            return {
                ...state,
                text: ''
            };
        default:
            return state;
    }
};

export default positionsFilterReducer;
