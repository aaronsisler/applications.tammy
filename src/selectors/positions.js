// Get filtered positions
export default (positions, { text = '' }) => positions
        .filter((position) => {
            const titleMatch = position.title.toLowerCase().includes(text.toLowerCase());
            const jobIdMatch = position.jobId.toLowerCase().includes(text.toLowerCase());
            return jobIdMatch || titleMatch;
        })
