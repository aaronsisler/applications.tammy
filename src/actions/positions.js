import database from '../firebase/firebase';
import { setPositions } from 'Actions/helpers/positions';

export const startSetPositions = () => (dispatch) =>
    database.ref(`positions`).once('value').then((snapshot) => {
        const positions = [];
        snapshot.forEach((childSnapshot) => {
            positions.push({
                positionId: childSnapshot.key,
                ...childSnapshot.val(),
            });
        });

        dispatch(setPositions(positions));
    });
