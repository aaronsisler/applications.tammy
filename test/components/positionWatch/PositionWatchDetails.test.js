import React from 'react';
import { shallow } from 'enzyme';
import { PositionWatchDetails } from 'PositionWatch/PositionWatchDetails';
import { position, positionId } from '../../fixtures/positions';

import history from 'Tools/history';
jest.mock('Tools/history');

describe('PositionWatchDetails', () => {
    const startSetWorkflowPosition = jest.fn();
    let wrapper;

    const buildWrapper = (positionIdInput, positionInput) => {
        wrapper = shallow(
            <PositionWatchDetails
                position={positionInput}
                positionId={positionIdInput}
                startSetWorkflowPosition={startSetWorkflowPosition}
            />
        );
    };

    describe('when positionId is available', () => {
        describe('when position is available', () => {
            const push = jest.fn();
            beforeEach(() => {
                history.push = push
                buildWrapper(positionId, position);
            });

            it('should render PositionWatchDetails correctly', () => {
                expect(wrapper).toMatchSnapshot();
            });

            describe('when button is clicked', () => {
                it('should navigate to /dashboard', () => {
                    wrapper.find('button').simulate('click');

                    expect(push).toHaveBeenLastCalledWith('/dashboard');
                });
            });
        });

        describe('when position is NOT available', () => {
            beforeEach(() => {
                buildWrapper(positionId, undefined);
            });

            it('should render PositionWatchDetails correctly', () => {
                expect(wrapper).toMatchSnapshot();
            });
        });
    });

    describe('when positionId is NOT available', () => {
        it('should render PositionWatchDetails correctly', () => {
            buildWrapper(undefined, undefined);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
