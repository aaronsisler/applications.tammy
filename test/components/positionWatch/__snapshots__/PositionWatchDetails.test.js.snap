// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PositionWatchDetails when positionId is NOT available should render correctly 1`] = `
<div
  className="inbox_details__empty inbox_mobile"
>
  Please select an item to view
</div>
`;

exports[`PositionWatchDetails when positionId is available when position is NOT available should render correctly 1`] = `
<div
  className="inbox_details__empty inbox_mobile"
>
  Position is not currently being watched. Please make sure to add a watch.
</div>
`;

exports[`PositionWatchDetails when positionId is available when position is available should render correctly 1`] = `
<div
  className="inbox_details"
>
  <div
    className="inbox_details_header"
  >
    <div
      className="inbox_details_header__actions"
    >
      <button
        className="inbox_details_header__mobile_button"
        onClick={[Function]}
      >
        Back to List
      </button>
      <LinkWrapper
        linkText="View Applicants"
        to="/applicants/firstPositionId"
      />
    </div>
    <div
      className="inbox_details_header__content"
    >
      <div
        className="inbox_details_header__title"
      >
        first title
      </div>
      <div
        className="inbox_details_header__job_id"
      >
        Job Id: 
        first job id
      </div>
      <div
        className="inbox_details_header__location"
      >
        Location: 
        first location
      </div>
    </div>
  </div>
  <div
    className="inbox_details_content"
  >
    <Connect(PositionWatchEditWidget)
      positionId="firstPositionId"
    />
    <PositionDetailsContent
      position={
        Object {
          "content": Object {
            "description": "first description",
            "requirements": Array [
              "Skill 1",
              "Skill 2",
            ],
            "responsibilities": Array [
              "Responsibility 1",
              "Responsibility 2",
            ],
          },
          "jobId": "first job id",
          "location": "first location",
          "positionId": "firstPositionId",
          "title": "first title",
        }
      }
    />
  </div>
</div>
`;
